src/ImageSharp/Formats/Jpeg/Components/Block8x8F.cs

```diff
@@ -485,6 +485,57 @@ public static unsafe void DequantizeBlock(Block8x8F* blockPtr, Block8x8F* qtPtr,
         /// <param name="source">The source block.</param>
         public static unsafe void Scale16X16To8X8(ref Block8x8F destination, ReadOnlySpan<Block8x8F> source)
         {
+#if SUPPORTS_RUNTIME_INTRINSICS
+            if (Avx2.IsSupported)
+            {
+                Scale16X16To8X8Vectorized(ref destination, source);
+                return;
+            }
+#endif
+
+            Scale16X16To8X8Scalar(ref destination, source);
+        }
+
+        private static void Scale16X16To8X8Vectorized(ref Block8x8F destination, ReadOnlySpan<Block8x8F> source)
+        {
+#if SUPPORTS_RUNTIME_INTRINSICS
+            Debug.Assert(Avx2.IsSupported, "AVX2 is required to execute this method");
+
+            var f2 = Vector256.Create(2f);
+            var f025 = Vector256.Create(0.25f);
+            Vector256<int> switchInnerDoubleWords = Unsafe.As<byte, Vector256<int>>(ref MemoryMarshal.GetReference(SimdUtils.HwIntrinsics.PermuteMaskSwitchInnerDWords8x32));
+
+            ref Vector256<float> in1 = ref Unsafe.As<Block8x8F, Vector256<float>>(ref MemoryMarshal.GetReference(source));
+            ref Vector256<float> in2 = ref Unsafe.As<Block8x8F, Vector256<float>>(ref Unsafe.Add(ref MemoryMarshal.GetReference(source), 1));
+            ref Vector256<float> destRef = ref Unsafe.As<Block8x8F, Vector256<float>>(ref destination);
+
+            for (int i = 0; i < 8; i++)
+            {
+                Vector256<float> a = in1;
+                Vector256<float> b = Unsafe.Add(ref in1, 1);
+                Vector256<float> c = in2;
+                Vector256<float> d = Unsafe.Add(ref in2, 1);
+
+                Vector256<float> calc1 = Avx.Shuffle(a, c, 0b10_00_10_00);
```

Comment:
I wish there was a way to produce constants via `SimdUtils.Shuffle.MmShuffle()` in a way that could be inlined.